src/mapml.js

```diff
@@ -3132,53 +3450,57 @@ M.templatedTileLayer = function(template, options) {
 M.MapMLStaticTileLayer = L.GridLayer.extend({
 
   initialize: function (options) {
-    let zoomBounds = this._getZoomBounds(options.tileContainer,options.maxZoomBound);
-    options.maxNativeZoom = zoomBounds.nMax;
-    options.minNativeZoom = zoomBounds.nMin;
-    options.maxZoom = zoomBounds.max;
-    options.minZoom = zoomBounds.min;
+    this.zoomBounds = this._getZoomBounds(options.tileContainer,options.maxZoomBound);
+    L.extend(options, this.zoomBounds);
     L.setOptions(this, options);
     this._groups = this._groupTiles(this.options.tileContainer.getElementsByTagName('tile'));
   },
 
   onAdd: function(){
-    this._bounds = this._getLayerBounds(this._groups,this._map.options.crs.options.resolutions); //stores meter values of bounds
-    this.isVisible = this._withinBounds(this._map.getPixelBounds(), this._bounds[this._map.getZoom()],this._map.options.crs.options.resolutions,this._map.getZoom());
+    this._bounds = this._getLayerBounds(this._groups,this._map.options.projection); //stores meter values of bounds
+    this.layerBounds = this._bounds[Object.keys(this._bounds)[0]];
+    for(let key of Object.keys(this._bounds)){
+      this.layerBounds.extend(this._bounds[key].min);
+      this.layerBounds.extend(this._bounds[key].max);
+    }
     L.GridLayer.prototype.onAdd.call(this,this._map);
+    this._map.fire('moveend',true);
```

Comment:
Why is this not an  invocation of parent moveend handler like you do below?  Is it a relic?

Reply:
This allows the newly added layer to fire the moveend handlers of the map including the handler of MapMLLayerControl, the layer responsible for disabling the controls in the panel. This allows a layer to be instantly disabled if its selected while out of bounds.